modules/core/src/main/scala/schema.scala

```diff
@@ -950,11 +950,45 @@ object SchemaValidator {
   def validateSchema(namedTypes: Result[List[NamedType]], defns: List[TypeDefinition]): Result[List[NamedType]] = {
     val undefinedResults = checkForUndefined(checkForDuplicates(namedTypes), defns)
 
-    val duplicates = NonEmptyChain.fromSeq(checkForEnumValueDuplicates(defns))
+    val errors = NonEmptyChain.fromSeq(checkForEnumValueDuplicates(defns) ++ validateImpls(defns))
+    errors.map(addLeft(undefinedResults, _)).getOrElse(undefinedResults)
+  }
+
+  def validateImpls(definitions: List[TypeDefinition]): List[Json] = {
+    val interfaces = definitions.collect[InterfaceTypeDefinition] {
+      case a: InterfaceTypeDefinition => a
+    }
+
+    val objects = definitions.collect[ObjectTypeDefinition] {
+      case a: ObjectTypeDefinition => a
+    }
+
+    objects.flatMap { obj =>
+      obj.interfaces.flatMap { ifaceName =>
+        interfaces.find(_.name == ifaceName.astName) match {
+          case Some(interface) => checkImplementation(obj, interface)
+          case None => List(mkError(s"Interface ${ifaceName.astName.value} implemented by object ${obj.name.value} is not defined"))
+        }
+      }
+    }
+  }
+
+  def checkImplementation(obj: ObjectTypeDefinition, interface: InterfaceTypeDefinition): List[Json] = {
+    interface.fields.flatMap { ifaceField =>
+      obj.fields.find(_.name == ifaceField.name).map { matching =>
+        if (ifaceField.tpe != matching.tpe) {
```

Comment:
Should these be compared with `=:=`?

Reply:
Do you mean `=!=`?

Reply:
So none of these types have an `Eq` instance at present so far as I can see? Happy to implement one if you reckon this is the way to go

Reply:
Sorry, `Type` specifically has an `=:=` operator that's like `==` but compares dealiases the types before comparing them. I don't know if it matters here. @milessabin will.

Reply:
Yes, you should compare types with `=:=` in the typical case.

Reply:
These are `Ast.Type`s though, not `Schema.Type`s, so there doesn't seem to be `=:=` defined unless I'm missing something?

Reply:
You're absolutely right! Yes, this is correct just as it is.